src/utils/array-utils.spec.js

```diff
@@ -0,0 +1,95 @@
+      const input = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9];
+      const result = swapItemsAtAt(input, 1, 10);
+      expect(result).toBe(input);
+    });
+    it('returns the original array if one of the indices are identical', () => {
+      const input = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9];
+      const result = swapItemsAtAt(input, 4, 4);
+      expect(result).toBe(input);
+    });
+  });
+
+  describe('removeItemAt', () => {
+    it('does not mutate the original array', () => {
+      const input = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9];
+      const result = removeItemAtAt(input, 4);
+      expect(result).not.toBe(input);
+      expect(input).toEqual([0, 1, 2, 3, 4, 5, 6, 7, 8, 9]);
+    });
+    it('removes the item on the correct index', () => {
+      const input = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9];
+      const result = removeItemAtAt(input, 4);
+      expect(result).toEqual([0, 1, 2, 3, 5, 6, 7, 8, 9]);
+    });
+    it('returns the original array if the index is out of range', () => {
+      const input = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9];
+      const result = removeItemAtAt(input, 10);
+      expect(result).toBe(input);
+    });
+  });
+
+  describe('ensureIsIncluded', () => {
+    it('returns the original array if the item was already included in the original array', () => {
+      const input = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9];
+      const result = ensureIsIncluded(input, 4);
+      expect(result).toBe(input);
+    });
+    it('returns a new array with the item added if the item was not already included in the original array', () => {
+      const input = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9];
+      const result = ensureIsIncluded(input, 10);
+      expect(result).toEqual([0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10]);
+    });
+    it('does not mutate the original array if the item was not already included in the original array', () => {
+      const input = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9];
+      const result = ensureIsIncluded(input, 10);
+      expect(result).not.toBe(input);
+      expect(input).toEqual([0, 1, 2, 3, 4, 5, 6, 7, 8, 9]);
+    });
+  });
+
+  describe('ensureIsExcluded', () => {
+    it('returns the original array if the item was not included anyway in the original array', () => {
+      const input = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9];
+      const result = ensureIsExcluded(input, 10);
+      expect(result).toBe(input);
+    });
+    it('returns a new array with the item removed if the item was included in the original array', () => {
+      const input = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9];
+      const result = ensureIsExcluded(input, 4);
+      expect(result).toEqual([0, 1, 2, 3, 5, 6, 7, 8, 9]);
+    });
+    it('does not mutate the original array if the item was included in the original array', () => {
+      const input = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9];
+      const result = ensureIsExcluded(input, 4);
+      expect(result).not.toBe(input);
+      expect(input).toEqual([0, 1, 2, 3, 4, 5, 6, 7, 8, 9]);
+    });
+  });
+
+  describe('shuffleItems', () => {
```

Comment:
We could even add a nice check if it is shuffled :)

For example we could do 
```
 expect(input.join()).not.toEqual(result.join());
```
And if you're considering maybe likelihood of a shuffle with exact output as original, we could shuffle twice and do something like 
```
expect(input.join() !== result1.join() || input.join() !== result2.join()).toBe(true);
```

What do you think?

Reply:
The second variant might work in a lot more cases, but it's in theory an unstable test.